decimal_test.go

```diff
@@ -388,28 +388,154 @@ func TestFloor(t *testing.T) {
 	}
 }
 
-func TestToStandard(t *testing.T) {
-	tests := map[string]string{
-		"0":           "0",
-		"0.0":         "0.0",
-		"0E2":         "000",
-		"0E-2":        "0.00",
-		"123.456E10":  "1234560000000",
-		".9":          "0.9",
-		"-.9":         "-0.9",
-		"-123.456E10": "-1234560000000",
-		"-0E-2":       "-0.00",
-	}
-
-	for c, r := range tests {
-		t.Run(c, func(t *testing.T) {
-			d, _, err := NewFromString(c)
+func TestFormat(t *testing.T) {
+	tests := map[string]struct {
+		e, E, f, g, G string
+	}{
+		"NaN":       {},
+		"Infinity":  {},
+		"-Infinity": {},
+		"sNaN":      {},
+		"0": {
+			e: "0e+0",
+			E: "0E+0",
+		},
+		"-0": {
+			e: "-0e+0",
+			E: "-0E+0",
+		},
+		"0.0": {
+			e: "0e-1",
+			E: "0E-1",
+		},
+		"-0.0": {
+			e: "-0e-1",
+			E: "-0E-1",
+		},
+		"0E+2": {
+			e: "0e+2",
+			f: "000",
+			g: "0e+2",
+		},
+	}
+	verbs := []string{"e", "E", "f", "g", "G"}
```

Comment:
to make linters happy, please consider using `verbs := []string{"%e", "%E", ...}` here and `Sprintf(v, d)` below.